lib/reducers/bwallet.js

```diff
@@ -3,14 +3,50 @@
  * maintain shared state for this plugin
  */
 
-const initialState = {};
+const {
+  USER_SELECT_WALLET,
+  USER_SELECT_ACCOUNT,
+  SELECTED_WALLET_NAMESPACE,
+  SELECTED_ACCOUNT_NAMESPACE,
+  SELECTED_PROPOSAL_NAMESPACE,
+  USER_SELECT_MULTISIG_PROPOSAL,
+} = require('../constants');
+
+const { reducersLogger: logger } = require('../helpers');
+
+const initialState = {
+  [SELECTED_WALLET_NAMESPACE]: null,
+  [SELECTED_ACCOUNT_NAMESPACE]: null,
+  [SELECTED_PROPOSAL_NAMESPACE]: null,
+};
 
 export default function bwalletReducer(state = initialState, action) {
-  // eslint-disable-next-line no-unused-vars
+  const newState = { ...state };
   const { type, payload } = action;
 
+  logger.debug(`[bwallet] type: ${type}`);
+
+  const {
+    walletId,
+    accountId,
+    proposalId,
+  } = payload;
+
   switch (type) {
+    case USER_SELECT_WALLET:
+      newState[SELECTED_WALLET_NAMESPACE] = walletId;
```

Comment:
Ah ok. So these are still subordinate to the plugins.bwallet store, you're just storing the key as a constant.